resolving/npm-resolver/src/index.ts

```diff
@@ -244,6 +256,64 @@ async function resolveNpm (
   }
 }
 
+async function tryResolveFromJsr (
+  ctx: ResolveFromNpmContext,
+  wantedDependency: WantedDependency,
+  opts: Omit<ResolveFromNpmOptions, 'registry'>,
+  defaultTag: string
+): Promise<ResolveResult | null> {
+  if (!wantedDependency.pref?.startsWith('jsr:')) return null
+
+  const registry = ctx.authConfig?.['@jsr:registry']
+  if (!registry) {
+    // Q: Why throw an error instead of falling back to a default URL?
+    // A: The packages from JSR registry may have dependencies in the `@jsr` scope which would
+    //    cause confusing error messages should it not be defined.
+
+    throw new PnpmError('JSR_REGISTRY_NOT_DEFINED', 'Cannot use JSR registry without defining a registry for the `@jsr` scope', {
+      hint: 'Fix it by adding `@jsr:registry=https://npm.jsr.io` to .npmrc',
+    })
+  }
```

Comment:
The jsr scope is empty in the npm registry: https://www.npmjs.com/~jsr

So, I wonder if we should change the default registry for the jsr scope. Although this could be considered a breaking change as other private registries could have that scope defined.

Reply:
If it's a breaking change, then it can be implemented in pnpm v11.

Reply:
> So, I wonder if we should change the default registry for the jsr scope. Although this could be considered a breaking change as other private registries could have that scope defined.

That would be really excellent honestly!